### PR TITLE
aws-sdk-cpp: 1.9.294 -> 1.10.2

### PR DESCRIPTION
**Pull request by tobim**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-burns** [comment]
Can you submit the patch upstream? Then we can use fetchpatch on it

**Comment by tobim** [comment]
https://github.com/NixOS/nix/pull/7294

**Comment by SuperSandro2000** [comment]
I've added a passthru change from #189133

File: pkgs/tools/package-management/nix/default.nix [hunk 1]
```diff
@@ -29,6 +29,11 @@ let
     requiredSystemFeatures = null;
   });
 
+  libstore-link-to-aws-crt-cpp-patch = fetchpatch {
+    url = "https://github.com/NixOS/nix/pull/7294/commits/07f2cb1e8f03784041475c27c2ba0aac7be6c0b7";
```
Comment (r-burns) [inline pkgs/tools/package-management/nix/default.nix:33]:
```suggestion
    url = "https://github.com/NixOS/nix/pull/7294/commits/07f2cb1e8f03784041475c27c2ba0aac7be6c0b7.patch";
```
and update sha256

Comment (r-burns) [reply pkgs/tools/package-management/nix/default.nix]: Please update the sha256 as well

**Comment by NickCao** [comment]
Superseded by https://github.com/NixOS/nixpkgs/pull/225512

**Pull request closed without merging**